app/code/BodenkoVV/AskQuestion/Ui/DataProvider.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace BodenkoVV\AskQuestion\Ui\Component;
+
+use Magento\Framework\Api\Filter;
+use Magento\Framework\Api\FilterBuilder;
+use Magento\Framework\Api\Search\SearchCriteriaBuilder;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\AuthorizationInterface;
+use Magento\Framework\View\Element\UiComponent\DataProvider\Reporting;
+
+class DataProvider extends \Magento\Framework\View\Element\UiComponent\DataProvider\DataProvider
+{
+    /**
+     * @var AuthorizationInterface
+     */
+    private $authorization;
+
+    /**
+     * @var AddFilterInterface[]
+     */
+    private $additionalFilterPool;
+
+    /**
+     * @param string $name
+     * @param string $primaryFieldName
+     * @param string $requestFieldName
+     * @param Reporting $reporting
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param RequestInterface $request
+     * @param FilterBuilder $filterBuilder
+     * @param array $meta
+     * @param array $data
+     * @param array $additionalFilterPool
+     * @SuppressWarnings(PHPMD.ExcessiveParameterList)
+     */
+    public function __construct(
+        $name,
+        $primaryFieldName,
+        $requestFieldName,
+        Reporting $reporting,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        RequestInterface $request,
+        FilterBuilder $filterBuilder,
+        array $meta = [],
+        array $data = [],
+        array $additionalFilterPool = []
+    ) {
+        parent::__construct(
+            $name,
+            $primaryFieldName,
+            $requestFieldName,
+            $reporting,
+            $searchCriteriaBuilder,
+            $request,
+            $filterBuilder,
+            $meta,
+            $data
+        );
+
+        $this->meta = array_replace_recursive($meta, $this->prepareMetadata());
+        $this->additionalFilterPool = $additionalFilterPool;
+    }
+
+    /**
+     * @deprecated 101.0.7
```

Comment:
Deprecated copied method is definitely not needed, please remove

Reply:
+